server/methods/accounts/accounts.app-test.js

```diff
@@ -504,76 +505,179 @@ describe("Account Meteor method ", function () {
   });
 
   describe("accounts/inviteShopMember", function () {
-    it("should not let non-Owners invite a user to the shop", function () {
-      sandbox.stub(Reaction, "hasPermission", () => false);
-      const createUserSpy = sandbox.spy(MeteorAccount, "createUser");
-      // create user
-      expect(() =>
-        Meteor.call("accounts/inviteShopMember", {
-          shopId,
-          groupId: Random.id(),
-          email: fakeUser.emails[0].address,
-          name: fakeUser.profile.addressBook[0].fullName
-        })).to.throw(Meteor.Error, /Access denied/);
-      // expect that createUser shouldnt have run
+    let createUserSpy, sendEmailSpy;
+    let groupId, group;
+
+    function callDescribed(accountAttributes = {}) {
+      const options = Object.assign({
+        shopId,
+        groupId,
+        email: fakeUser.emails[0].address,
+        name: fakeUser.profile.addressBook[0].fullName
+      }, accountAttributes);
+
+      return Meteor.call("accounts/inviteShopMember", options);
+    }
+
+    function stubPermissioning(settings) {
+      const { hasPermission, canInviteToGroup } = settings;
+
+      sandbox.stub(Reaction, "hasPermission", () => hasPermission);
+      sandbox
+        .stub(Reaction, "canInviteToGroup", () => canInviteToGroup)
+        .withArgs({ group, user: fakeUser });
+    }
+
+    beforeEach(function () {
+      createUserSpy = sandbox.spy(MeteorAccounts, "createUser");
+      sendEmailSpy = sandbox.stub(Reaction.Email, "send"); // stub instead of spy so we don't actually try to send
+
+      sandbox.stub(Meteor, "userId", () => fakeUser.userId);
+      sandbox.stub(Meteor, "user", () => fakeUser);
+
+      groupId = Random.id();
+      group = Factory.create("group");
+      sandbox.stub(Groups, "findOne", () => group).withArgs({ _id: groupId });
+    });
+
+    it("requires reaction-accounts permission", function () {
+      stubPermissioning({ hasPermission: false });
+      sandbox.stub(Logger, "error") // since we expect this, let's keep the output clean
+        .withArgs(sinon.match(/reaction-accounts permissions/));
+
+      expect(callDescribed).to.throw(Meteor.Error, /Access denied/);
       expect(createUserSpy).to.not.have.been.called;
     });
 
-    it("should let a Owner invite a user to the shop", function (done) {
-      this.timeout(20000);
-      this.retries(3);
-      sandbox.stub(Reaction, "hasPermission", () => true);
-      // TODO: Need to udpate this test to properly check the account created
-      // there's currently an error with Media branding asset when trying to do that
-      expect(() =>
-        Meteor.call("accounts/inviteShopMember", {
-          shopId,
-          groupId: Random.id(),
-          email: fakeUser.emails[0].address,
-          name: fakeUser.profile.addressBook[0].fullName
-        })).to.not.throw(Meteor.Error, /Access denied/);
```

Comment:
several of the expectations did not exactly match the test descriptions, so I re-wrote the whole describe block for `inviteShopMember` and `inviteShopOwner`.